Products/ZenMessaging/queuemessaging/publisher.py

```diff
@@ -232,8 +234,10 @@ def getModelChangePublisher():
         tx._synchronizedPublisher = ModelChangePublisher()
         # create new PublishSynchronizer also add after completion hook so that client/channel can be closed
         pSync = PublishSynchronizer()
-        tx.addBeforeCommitHook(pSync.beforeCompletionHook, [tx])
-        tx.addAfterCommitHook(pSync.afterCompletionHook, [tx])
+        with BEFORE_COMMIT_HOOK_TIMER:
+            tx.addBeforeCommitHook(pSync.beforeCompletionHook, [tx])
```

Comment:
I believe this only causes us to measure how long it takes to add the hook. Not how long the hooks take to execute on each commit.

Reply:
Added the timers as a decorator to `beforeCompletionHook` and `afterCompletionHook` functions.